Analytics/Classes/Internal/SEGHTTPClient.m

```diff
@@ -153,6 +160,13 @@ - (NSURLSessionDataTask *)settingsForWriteKey:(NSString *)writeKey completionHan
 
         NSInteger code = ((NSHTTPURLResponse *)response).statusCode;
         if (code > 300) {
+            if(response && data && code < 500 && ![self.cache cachedResponseForRequest:request]) {
+                // Cache 4xx responses
+                NSCachedURLResponse* cached = [NSCachedURLResponse.alloc initWithResponse:response
+                                                                                     data:data];
+                [self.cache storeCachedResponse:cached forRequest:request];
```

Comment:
how long is ths cached for?

Reply:
It'll only be one hour (according to `max-age`) and after it'll start issuing a `HEAD` request to check `ETag`. We can also store it indefinitely if needed